splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/WithStatementIT.java

```diff
@@ -209,5 +231,34 @@ public void testInsertContainingWith() throws Exception {
         assertEquals(expectedResult, TestUtils.FormattedResult.ResultFactory.toString(rs));
     }
 
+    @Test
+    public void testWithInSchemaNotSufficientPrivilege() throws Exception {
+        methodWatcher.executeUpdate("insert into t12 values (1)");
+        String expected = "I |\n" +
+                "----\n" +
+                " 1 |";
+
+        // user with enough privilege, dynamic view in SESSION schema
+        try (ResultSet rs = methodWatcher.executeQuery("with dt as (select * from t12) select * from dt")) {
+            assertEquals(expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        }
+        // user with enough privilege, dynamic view in current schema
+        try (ResultSet rs = methodWatcher.executeQuery(format("with %s.dt as (select * from t12) select * from dt", SCHEMA))) {
+            assertEquals(expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        }
+
+        try (ResultSet rs = testUserConn.createStatement().executeQuery("values current schema")) {
+            String expectedSchema = "1        |\n" +
+                    "-----------------\n" +
+                    "WITHSTATEMENTIT |";
+            assertEquals(expectedSchema, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        }
 
+        // user with no write privilege, dynamic view in SESSION schema
+        try (ResultSet rs = testUserConn.createStatement().executeQuery(format("with dt as (select * from t12) select * from dt"))) {
+            assertEquals(expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        }
+        // user with no write privilege, dynamic view in current schema
+        assertFailed(testUserConn, format("with %s.dt as (select * from t12) select * from dt", SCHEMA), SQLState.AUTH_NO_ACCESS_NOT_OWNER);
+    }
```

Comment:
Could you add a test case to test that in two concurrent connections, we can create two dynamic views using the same name under the session schema, and they will not interfere with each other? Thanks!

Reply:
Done.

Reply:
Thanks!